### PR TITLE
adding 3.7 to the tests

### PR DESCRIPTION
**Pull request by balopat**
To cover all python runtimes that we support, adding coverage for 3.7 besides 3.6.
With #3194 we can add 3.8 as well, and soon 3.9 hopefully to the mix. 
If we agree on #3347, then we can drop 3.6 as well.

### CODE REVIEW

File: Untitled.ipynb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
```
Comment (vtomole) [inline Untitled.ipynb:1]: Remove this file.

**Review by vtomole — CHANGES_REQUESTED** [review]

File: Untitled.ipynb (hunk 1 above)
Comment (balopat) [reply Untitled.ipynb:1]: ahh :D thanks.

File: requirements.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Runtime requirements for the python 3 version of cirq.
 
-dataclasses; python_version < '3.7'
```
Comment (vtomole) [inline requirements.txt:3]: Is this change relevant to this PR? I haven't been keeping track on if we've are doing `3.8`.

Comment (balopat) [reply requirements.txt:3]: This is PR is just raising the level to 3.7 i.e. < 3.8 should be fine. I've been using cirq on 3.7 for a long time now. The 3.8 work will build on top of this.

**Review by vtomole — APPROVED** [review]
LGTM
